py/legacypipe/format_catalog.py

```diff
@@ -236,24 +236,51 @@ def add_wiselike(c, bands=None):
             add_wiselike(c, bands=['w1','w2'])
         # Cut down to a fixed number of WISE time-resolved epochs?
         if N_wise_epochs is not None:
-            for col in lc_cols:
-                for band in ['w1','w2']:
+
+            # mapping from (nobs, mjd) to indices to keep
+            keep_epochs = {}
+            newvals = {}
+
+            for band in ['w1', 'w2']:
+                # initialize new (cut) arrays
+                for col in lc_cols:
                     colname = col + '_' + band
-                    # Cut or pad old value to have N_wise_epochs-length arrays
                     oldval = T.get(colname)
                     n,ne = oldval.shape
-                    newval = np.zeros((n,N_wise_epochs), oldval.dtype)
-                    ncopy = min(N_wise_epochs,ne)
-                    newval[:, :ncopy] = oldval[:,:ncopy]
-                    T.set(colname, newval)
+                    newval = np.zeros((n, N_wise_epochs), oldval.dtype)
+                    newvals[colname] = newval
+
+                lc_nobs = T.get('lc_nobs_%s' % band)
+                lc_mjd = T.get('lc_mjd_%s' % band)
+                # Check each row (source) individually, since coverage isn't
+                # uniform across a brick
+                for row,(nobs,mjd) in enumerate(zip(lc_nobs, lc_mjd)):
+                    key = tuple(nobs) + tuple(mjd)
+                    if not key in keep_epochs:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [test for membership should be 'not in'](https://app.codacy.com/app/dstndstn/legacypipe/pullRequest?prid=4730106)